src/sentry/services/hybrid_cloud/rpc.py

```diff
@@ -484,6 +484,7 @@ def _send_to_remote_silo(self, use_test_client: bool) -> Any:
             tags=self._metrics_tags(),
         )
         with self._open_request_context():
+            self._check_disabled()
```

Comment:
I tried to keep this as close to firing requests as possible so that when RPCs are disabled we'll still get metrics, that might either be a good idea or a bad idea depending on why we're disabling them